homeassistant/components/device_tracker/__init__.py

```diff
@@ -205,7 +206,16 @@ def async_see_service(call):
         """Service to see a device."""
         args = {key: value for key, value in call.data.items() if key in
                 (ATTR_MAC, ATTR_DEV_ID, ATTR_HOST_NAME, ATTR_LOCATION_NAME,
-                 ATTR_GPS, ATTR_GPS_ACCURACY, ATTR_BATTERY, ATTR_ATTRIBUTES)}
+                 ATTR_GPS, ATTR_GPS_ACCURACY, ATTR_BATTERY, ATTR_ATTRIBUTES,
+                 ATTR_SOURCE_TYPE)}
+
+        try:
+            consider_home = call.data[ATTR_CONSIDER_HOME]
+        except KeyError:
```

Comment:
Please use `dict.get` and set the default timedelta if the key is missing.

Reply:
The other arguments/keys are only added if they are present in call.data, so what is the reason you prefer a default for `consider_home`? This would overwrite any `consider_home` stored in a `Device` already. If `consider_home` isn't passed, then the value previously stored in `Device.consider_home` is used (see `Device.async_seen`).

If your issue is solely with the try/except block, I can of course rewrite it differently, e.g.
```
if ATTR_CONSIDER_HOME in call.data:
    args[ATTR_CONSIDER_HOME] = timedelta(seconds=int(call.data[ATTR_CONSIDER_HOME]))
```

Reply:
Yes, sorry, I misread the try... else. Just switch from try... else to what you write above.

Reply:
Alright. Commits are squashed before merging right? So I don't have to force-push to my fork?

One more thing might be to not just allow seconds in the timedelta creation, but to parse times such as "0:03", "0:03:00" correctly.

Reply:
Yes, instead add a service schema that validates all the allowed key - value pairs. Then you can also just add the `ATTR_CONSIDER_HOME` key in the iteration of `call.data` above.

Reply:
That's awesome, I didn't know that possibility existed.